frontend/src/components/features/TaakFeatureConfiguration.tsx

```diff
@@ -0,0 +1,115 @@
+import {ChangeEvent, useEffect, useState} from "react";
+import FeatureConfigurationProps from "../../interfaces/FeatureConfigurationProps.ts";
+import Fieldset, {FieldsetLegend} from "@gemeente-denhaag/form-fieldset";
+import {FormattedMessage} from "react-intl";
+import {Heading3, Paragraph} from "@gemeente-denhaag/typography";
+import {FormField} from "@gemeente-denhaag/form-field";
+import {Checkbox} from "@gemeente-denhaag/checkbox";
+import {FormLabel} from "@gemeente-denhaag/form-label";
+import styles from '../../pages/FeatureConfigurationPage.module.scss'
+import TextInput from "@gemeente-denhaag/text-input";
+import _ from "lodash";
+
+interface TaakConfiguration {
+  enabled?: string;
+  properties?: {
+    "taakobject": {
+      "type-url"?: string,
+      "type-url-v2"?: string
+    }
+  }
+}
+
+interface TaakFeatureConfigurationProps extends FeatureConfigurationProps {
+  featureConfiguration: TaakConfiguration | undefined
+}
+
+const TaakFeatureConfiguration = ({
+                                    featureConfiguration,
+                                    onValid,
+                                    onChange
+                                  }: TaakFeatureConfigurationProps) => {
+  const [taakConfiguration, setTaakConfiguration] = useState<TaakConfiguration | undefined>()
+
+  useEffect(() => {
+    if (featureConfiguration) setTaakConfiguration(featureConfiguration)
+  }, [featureConfiguration]);
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(taakConfiguration);
+    }
+  }, [taakConfiguration, onChange]);
+
+  useEffect(() => {
+    if (onValid) onValid(true)
+  }, [onValid]);
+
+  const handleInputChange = (
+    propertyPath: string,
+    event: ChangeEvent<HTMLInputElement>
+  ) => {
+    const newConfiguration = _.clone(taakConfiguration ?? {});
+    _.set(newConfiguration, propertyPath, event.target.value)
+    setTaakConfiguration(newConfiguration)
+  }
+
+  return (
+    <Fieldset className={styles["feature-config__form"]}>
+      <FieldsetLegend className="utrecht-form-fieldset__legend--distanced">
+        <Heading3><FormattedMessage id={"features.taak.configuration"}></FormattedMessage></Heading3>
+      </FieldsetLegend>
+      <FormField type="checkbox">
+        <Paragraph className="utrecht-form-field__label">
+          <Checkbox
+            id="taak.enabled"
+            className="utrecht-form-field__input"
+            checked={taakConfiguration?.enabled == "true"}
+            onChange={(e) => {
+              setTaakConfiguration({
+                ...taakConfiguration,
+                enabled: e.target.checked.toString(),
+              })
+            }}
+          />
+          <FormLabel htmlFor="true" type="checkbox">
+            <FormattedMessage id={"features.taak.enable"}></FormattedMessage>
+          </FormLabel>
+        </Paragraph>
+      </FormField>
+      {taakConfiguration?.enabled == "true" && (
+        <>
```

Comment:
This should not be needed since the function component always returns something regardless of this flow.

Reply:
When I remove the <> I get compile errors right away. Which is logical because I add an extra two FormFields.

Reply:
Ah, I see why. It's because thiss has more than one element being returned by the conditional function. My bad.

Could you please use the full `<Fragment>` though instead of this shorthand, while it's the same thing I feel like the full notation is more obvious